### ISSUE TITLE
Misleading error message for matplotlib.pyplot.errorbar()

### ISSUE THREAD

**Issue by dlmccaffrey** [matplotlib/matplotlib#6933]
I'm running on 64 bit Windows 7 with Spyder 2.3.8, Python 3.5.1, and Matplotlib 1.5.1. I've installed Visual C++ Redistributable for Visual Studio 2015.

I ran into an interesting issue with the matplotlib.pyplot.errorbar() function. I didn't read the documentation thoroughly and assumed that I would be able to use 2D arrays as inputs like with plot(). So, when I used the following code:
```
X=numpy.array([1,2,3])
Y=numpy.array([[1,5,2],[3,6,4],[9,3,7]])
Yerr=numpy.ones_like(Y)
matplotlib.pyplot.errorbar(X, Y, Yerr)
```
I get the following error:
```
File "\lib\site-packages\matplotlib\axes\_axes.py", line 2964, in errorbar
    raise ValueError("yerr must be a scalar, the same "

ValueError: yerr must be a scalar, the same dimensions as y, or 2xN.
```
I suggest changing the message to something like "yerr must be [ scalar | N, Nx1, or 2xN array-like ]" to match the documentation. Same for xerr.

Consequently, is there a reason that errorbar() needs flattened data when plot() doesn't? It might be worth changing. I ran into the issue when I was using plot() to plot trendlines for multiple datasets and wanted to put errorbars on the data. Flattening the trendlines doesn't work well because it connects the end of one trendline to the beginning of the next.

**Comment by tacaswell** [comment]
That sounds like a reasonable change to the error message.  Can you open a pull request with that change?

I suspect the reason that errorbar does less internal broadcasting is the ambiguity between when the second dimension is to specify upper/lower or should be broadcast.  It also seems that there maybe some subtlety with dealing with the interaction between x, y, xerr, and yerr broadcasting.

It is a detailed analysis of what the broadcasting rules would have to be.  It is not clear to me if broadcasting is not already implemented because there is an unresolvable issue or if no one has taken the time to implement it.

@dlmccaffrey Would you be interested in taking a crack at the analysis/implementation?

**Comment by dlmccaffrey** [comment]
I can give it a shot. I've never collaborated on code or used GitHub before. Do I just take a branch, work on it, and submit a pull request when I'm done? Are there guidelines I should follow when I work on it?

**Comment by tacaswell** [comment]
Then this is a great time to start :smile: 

See http://matplotlib.org/devel/gitwash/git_development.html for detailed directions.

TL;DR

 - fork mpl on github
 - clone your fork locally
 - create a development branch
 - hack-commit-hack-...
 - push your branch to your mpl fork on github
 - create a pull request via the github website
 - get reviewed, iterate, and eventually get merged.

If you have any questions try the mailing list (matplotlbi-devel@python.org you will have to register to post un-moderated), gitter ( https://gitter.im/matplotlib/matplotlib ), or a comment here.

**Comment by dlmccaffrey** [comment]
So I thought I'd just change the error message to get started and followed the instructions for creating a feature branch from master. Since I branched from master, this is going to be a change in the new release and I should request a pull against master. For fixing it in 1.5.2, I need to branch from v1.5.2 and request a pull against v1.5.2. Correct?

**Comment by jenshnielsen** [comment]
It's perfectly fine to do the change against master and open a pull request against master. We can then take care of backporting it to the relevant branches when we merge it but you can also do it the way that you describe but targeting 2.x. (Because there is no 1.5.x branch)

**Comment by tacaswell** [comment]
@VHK459 If you think there is still a bug in Matplotlib, please open a new issue with a code snippet that reproduces the problem.  If you have a question about to use Matplotlib please open a new topic at https://discourse.matplotlib.org (again with an running example illustrates your question).
